apps/prairielearn/src/lib/github.js

```diff
@@ -245,291 +170,203 @@ module.exports = {
-            options.title,
-            options.display_timezone,
-            options.path,
-            `git@github.com:${config.githubCourseOwner}/${options.repo_short_name}.git`,
-            options.branch,
-            authn_user.user_id,
-          ];
-          inserted_course = (await sqldb.callAsync('courses_insert', sql_params)).rows[0];
-          job.verbose('Inserted course into database:');
-          job.verbose(JSON.stringify(inserted_course, null, 4));
-        },
-      );
+      logger.info('Adding course to database');
+      const inserted_course = (
+        await sqldb.callAsync('courses_insert', [
+          options.institution_id,
+          options.short_name,
+          options.title,
+          options.display_timezone,
+          options.path,
+          `git@github.com:${config.githubCourseOwner}/${options.repo_short_name}.git`,
+          branch,
+          authn_user.user_id,
+        ])
+      ).rows[0];
+      job.verbose('Inserted course into database:');
+      job.verbose(JSON.stringify(inserted_course, null, 4));
 
       // Give the owner required permissions
-      await module.exports._runJobAsync(
-        { type: 'update_', description: 'Giving user owner permission' },
-        job_sequence_id,
-        authn_user.user_id,
-        async () => {
-          const sql_params = {
-            course_id: inserted_course.id,
-            course_request_id: options.course_request_id,
-          };
-          await sqldb.queryOneRowAsync(sql.set_course_owner_permission, sql_params);
-        },
-      );
+      logger.info('Giving user owner permission');
+      await sqldb.queryOneRowAsync(sql.set_course_owner_permission, {
+        course_id: inserted_course.id,
+        course_request_id: options.course_request_id,
+      });
 
       // Automatically sync the new course. This part is shamelessly stolen
       // from `pages/shared/syncHelpers.js`.
       const git_env = process.env;
       if (config.gitSshCommand != null) {
         git_env.GIT_SSH_COMMAND = config.gitSshCommand;
       }
-      await module.exports._runJobCommandAsync(
-        {
-          type: 'clone_from_git',
-          description: 'Clone from remote git repository',
-          command: 'git',
-          arguments: ['clone', inserted_course.repository, inserted_course.path],
-          env: git_env,
-        },
-        job_sequence_id,
-        authn_user.user_id,
-      );
-      let sync_result;
-      await module.exports._runJobAsync(
-        {
-          type: 'sync_from_disk',
-          description: 'Sync git repository to database',
-          last_in_sequence: !config.chunksGenerator,
-        },
-        job_sequence_id,
-        authn_user.user_id,
-        async function (job) {
-          sync_result = await syncFromDisk.syncDiskToSqlAsync(
-            inserted_course.path,
-            inserted_course.id,
-            job,
-          );
-        },
+
+      logger.info('Clone from remote git repository');
```

Comment:
I believe this part (and a good chunk of the following code) could use the new `course.pullAndUpdate`.